analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.cs

```diff
@@ -315,6 +316,88 @@ public void GetHashCode_ReturnsSameForEquivalent()
             mixedOrig.GetHashCode().Should().NotBe(mixedDiff.GetHashCode());
         }
 
+        [TestMethod]
+        public void ToString_Empty() =>
+            ProgramState.Empty.ToString().Should().Be("Empty");
+
+        [TestMethod]
+        public void ToString_WithSymbols()
+        {
+            var assignment = TestHelper.CompileCfgBodyCS("var a = true;").Blocks[1].Operations[0];
+            var counter = new SymbolicValueCounter();
+            var variableSymbol = assignment.Children.First().TrackedSymbol();
+            var sut = ProgramState.Empty.SetSymbolValue(variableSymbol, null);
+            sut.ToString().Should().BeIgnoringLineEndings(
+@"Symbols:
+a: <null>
+");
+
+            sut = ProgramState.Empty.SetSymbolValue(variableSymbol, new SymbolicValue(counter));
+            sut.ToString().Should().BeIgnoringLineEndings(
+@"Symbols:
+a: SV_1
+");
+
+            var valueWithConstraint = new SymbolicValue(counter);
+            valueWithConstraint.SetConstraint(TestConstraint.Second);
+            sut = sut.SetSymbolValue(variableSymbol.ContainingSymbol, valueWithConstraint);
+            sut.ToString().Should().BeIgnoringLineEndings(
+@"Symbols:
+a: SV_1
+Sample.Main(): SV_2: Second
+");
+        }
+
+        [TestMethod]
+        public void ToString_WithOperations()
+        {
+            var assignment = TestHelper.CompileCfgBodyCS("var a = true;").Blocks[1].Operations[0];
+            var counter = new SymbolicValueCounter();
```

Comment:
We use a counter in all the tests. We could make it a field and use a TestInitialize method to construct it before each test.

Reply:
I'm considering refactoring these UTs to use `TestInitialize` to generate some `context` field that would provide `Operation1`, `Operation2`, `Operation3`, `Value1`, `Value2`, `Value3`, `Symbol1`, `Symbol2` and `Symbol3` as those repeat in most of the cases. The same type could be then used in `ProgramStateTest`. In another PR